File: app/src/constants/api_paths.ts
// ─── 认证 ─────────────────────────────────────────────────────────────────
export const AUTH_LOGIN = '/api/auth/login'
export const AUTH_REFRESH = '/api/auth/refresh'
export const AUTH_LOGOUT = '/api/auth/logout'
export const AUTH_ME = '/api/auth/me'
export const AUTH_CHANGE_PASSWORD = '/api/auth/change-password'
export const AUTH_FORGOT_PASSWORD = '/api/auth/forgot-password'
export const AUTH_RESET_PASSWORD = '/api/auth/reset-password'

// ─── 用户管理 ─────────────────────────────────────────────────────────────
export const USERS = '/api/users'

// ─── 组织架构 ─────────────────────────────────────────────────────────────
export const DEPARTMENTS = '/api/departments'

// ─── 资产 ─────────────────────────────────────────────────────────────────
export const BUILDINGS = '/api/buildings'
export const FLOORS = '/api/floors'
export const UNITS = '/api/units'
export const ASSETS_OVERVIEW = '/api/assets/overview'

// ─── 租务 ─────────────────────────────────────────────────────────────────
export const TENANTS = '/api/tenants'
export const CONTRACTS = '/api/contracts'

// ─── 财务 ─────────────────────────────────────────────────────────────────
export const INVOICES = '/api/invoices'
export const PAYMENTS = '/api/payments'
export const DEPOSITS = '/api/deposits'
export const EXPENSES = '/api/expenses'

// ─── 工单 ─────────────────────────────────────────────────────────────────
export const WORK_ORDERS = '/api/work-orders'
export const SUPPLIERS = '/api/suppliers'

// ─── 二房东 ───────────────────────────────────────────────────────────────
export const SUBLEASES = '/api/subleases'

// ─── 预警与通知 ───────────────────────────────────────────────────────────
export const ALERTS = '/api/alerts'
export const NOTIFICATIONS = '/api/notifications'

// ─── KPI ──────────────────────────────────────────────────────────────────
export const KPI_SCHEMES = '/api/kpi/schemes'
export const KPI_SCORES = '/api/kpi/scores'
export const KPI_APPEALS = '/api/kpi/appeals'

// ─── NOI & WALE ──────────────────────────────────────────────────────────
export const NOI = '/api/noi'
export const WALE = '/api/wale'

// ─── 水电抄表 ─────────────────────────────────────────────────────────────
export const METER_READINGS = '/api/meter-readings'

// ─── 商铺营业额 ──────────────────────────────────────────────────────────
export const TURNOVER_REPORTS = '/api/turnover-reports'

// ─── 文件代理 ─────────────────────────────────────────────────────────────
/** 后端文件代理根路径 */
export const FILES = '/api/files'

/**
 * 构造文件代理完整 URL（用于 SVG / PDF 等需要带 token 直接拉取的资源）
 *
 * @param storagePath 后端返回的相对路径（如 `floors/{building_id}/{floor_id}.svg`）
 * @returns 完整 URL（如 `http://host:port/api/files/floors/.../xxx.svg`）
 */
export function buildFileProxyUrl(storagePath: string): string {
  const baseUrl = (import.meta.env.VITE_API_BASE_URL as string | undefined) ?? ''
  // 去掉 storagePath 前导斜杠，避免双斜杠
  const path = storagePath.startsWith('/') ? storagePath.slice(1) : storagePath
  return `${baseUrl}${FILES}/${path}`
}

File: flutter_app/lib/core/api/api_paths.dart
/// All API path constants extracted from API_INVENTORY v1.7.
///
/// Usage: `ApiPaths.authLogin` → `/api/auth/login`
/// Never hardcode path strings in Repository or BLoC code.
abstract final class ApiPaths {
  // ── Auth ──
  static const authLogin = '/api/auth/login';
  static const authRefresh = '/api/auth/refresh';
  static const authLogout = '/api/auth/logout';
  static const authMe = '/api/auth/me';
  static const authChangePassword = '/api/auth/change-password';
  static const authForgotPassword = '/api/auth/forgot-password';
  static const authResetPassword = '/api/auth/reset-password';

  // ── Users ──
  static const users = '/api/users';

  // ── Departments ──
  static const departments = '/api/departments';
  static const managedScopes = '/api/managed-scopes';

  // ── Assets ──
  static const buildings = '/api/buildings';
  static const floors = '/api/floors';
  static const floorPlans = '/api/floor-plans';
  static const units = '/api/units';
  static const unitsImport = '/api/units/import';
  static const unitsExport = '/api/units/export';
  static const renovations = '/api/renovations';
  static const assetsOverview = '/api/assets/overview';

  // ── Tenants ──
  static const tenants = '/api/tenants';

  // ── Contracts ──
  static const contracts = '/api/contracts';
  static const contractsWale = '/api/contracts/wale';
  static const contractsWaleTrend = '/api/contracts/wale/trend';
  static const contractsWaleWaterfall = '/api/contracts/wale/waterfall';
  static const contractsWaleDashboard = '/api/contracts/wale/dashboard';
  static const contractsAtRisk = '/api/contracts/at-risk';

  // ── Escalation Templates ──
  static const escalationTemplates = '/api/escalation-templates';

  // ── Alerts ──
  static const alerts = '/api/alerts';
  static const alertsUnread = '/api/alerts/unread';

  // ── Deposits ──
  static const deposits = '/api/deposits';

  // ── Invoices ──
  static const invoices = '/api/invoices';
  static const invoicesGenerate = '/api/invoices/generate';
  static const invoicesExport = '/api/invoices/export';

  // ── Payments ──
  static const payments = '/api/payments';

  // ── Expenses ──
  static const expenses = '/api/expenses';

  // ── NOI ──
  static const noiSummary = '/api/noi/summary';
  static const noiTrend = '/api/noi/trend';
  static const noiBreakdown = '/api/noi/breakdown';
  static const noiVacancyLoss = '/api/noi/vacancy-loss';
  static const noiBudget = '/api/noi/budget';

  // ── Meter Readings ──
  static const meterReadings = '/api/meter-readings';
  static const meterReadingsAllocationPreview =
      '/api/meter-readings/allocation-preview';

  // ── Turnover Reports ──
  static const turnoverReports = '/api/turnover-reports';

  // ── Dunning ──
  static const dunningLogs = '/api/dunning-logs';

  // ── KPI ──
  static const kpiMetrics = '/api/kpi/metrics';
  static const kpiSchemes = '/api/kpi/schemes';
  static const kpiScores = '/api/kpi/scores';
  static const kpiScoresGenerate = '/api/kpi/scores/generate';
  static const kpiScoresRecalculate = '/api/kpi/scores/recalculate';
  static const kpiRankings = '/api/kpi/rankings';
  static const kpiTrends = '/api/kpi/trends';
  static const kpiExport = '/api/kpi/export';
  static const kpiAppeals = '/api/kpi/appeals';

  // ── Work Orders ──
  static const workorders = '/api/workorders';
  static const workordersCostReport = '/api/workorders/cost-report';
  static const suppliers = '/api/suppliers';

  // ── Subleases (Internal) ──
  static const subleases = '/api/subleases';
  static const subleasesDashboard = '/api/subleases/dashboard';
  static const subleasesExport = '/api/subleases/export';

  // ── Sublease Portal ──
  static const subleasePortalLogin = '/api/sublease-portal/login';
  static const subleasePortalUnits = '/api/sublease-portal/units';
  static const subleasePortalSubleases = '/api/sublease-portal/subleases';

  // ── Imports ──
  static const imports = '/api/imports';

  // ── Files ──
  static const files = '/api/files';

  // ── Audit Logs ──
  static const auditLogs = '/api/audit-logs';

  // ── Jobs ──
  static const jobsExecutions = '/api/jobs/executions';

  // ── Notifications ──
  static const notifications = '/api/notifications';
  static const notificationsUnreadCount = '/api/notifications/unread-count';

  // ── Health ──
  static const health = '/api/health';
}
